shared/src/business/useCases/updatePetitionerInformationInteractor.js

```diff
@@ -388,6 +389,38 @@ exports.updatePetitionerInformationInteractor = async ({
     }
   }
 
+  if (
+    contactPrimary.email &&
+    contactPrimary.email !== oldCase.contactPrimary.email
+  ) {
+    const isEmailAvailable = await applicationContext
+      .getPersistenceGateway()
+      .isEmailAvailable({
+        applicationContext,
+        email: contactPrimary.email,
+      });
+
+    if (isEmailAvailable) {
+      caseEntity = await applicationContext
+        .getUseCaseHelpers()
+        .createUserForContactPrimary({
+          applicationContext,
+          caseEntity,
+          email: contactPrimary.email,
+          name: contactPrimary.name,
+        });
```

Comment:
I must be missing something, so please help me understand 🙏. I don't see how this function changes the case Entity / contactPrimary of the Case entity.

Reply:
And it looks like we're not testing for this flow in the accompanying test file.

Reply:
I think near the top we run
```
let caseEntity = new Case(
    {
      ...oldCase,
      contactPrimary: {
        ...oldCase.contactPrimary,
        ...primaryEditableFields,
      },
      contactSecondary: {
        ...oldCase.contactSecondary,
        ...secondaryEditableFields,
      },
      partyType,
    },
    { applicationContext },
  );
  ```
  
  then we persist the case at the end
  ```
 const updatedCase = await applicationContext
    .getUseCaseHelpers()
    .updateCaseAndAssociations({
      applicationContext,
      caseToUpdate: caseEntity,
    });
    ```

Reply:
Test added!